### PR TITLE
add disableLocalStorage option

### PR DESCRIPTION
**Pull request by rotimi-best**
This feature implements disableLocalStorage prop just like in novel.sh.

Original implementation in novel.sh can be found [here](https://github.com/steven-tey/novel/commit/9e8379f16df7cf3c8ab68e61c8295a4f001cce32)

### CODE REVIEW

File: src/lib/ui/editor/index.svelte [hunk 1]
```diff
@@ -66,6 +66,11 @@
 	 * Defaults to "novel__content".
 	 */
 	export let storageKey = 'novel__content';
+	/**
+	 * Disable local storage read/save.
+	 * Defaults to false.
+	 */
+	export let disableLocalStorage = false;
```
Comment (TGlide) [inline src/lib/ui/editor/index.svelte:73]:
```suggestion
	/**
	 * Disable local storage read/save.
	 * @default false
	 */
	export let disableLocalStorage = false;
```

File: src/lib/ui/editor/index.svelte [hunk 2]
```diff
@@ -115,7 +125,11 @@
 
 	const debouncedUpdates = createDebouncedCallback(async ({ editor }) => {
 		const json = editor.getJSON();
-		content.set(json);
+
+		if (!disableLocalStorage) {
```
Comment (TGlide) [inline src/lib/ui/editor/index.svelte:129]: We don't need to get JSON if we're not setting anything. We can do an early return at the start of the callback:
`if (disableLocalStorage) return;`

Comment (rotimi-best) [reply src/lib/ui/editor/index.svelte:129]: @TGlide Don't we want to call `onDebouncedUpdate(editor)` on line 133?

does the function have a connection with localStorage?

Comment (TGlide) [reply src/lib/ui/editor/index.svelte:139]: @rotimi-best we probably do! But it's a separate point from what I mentioned

Comment (rotimi-best) [reply src/lib/ui/editor/index.svelte:139]: Oh I get what you mean. Will make the update.

Comment (rotimi-best) [reply src/lib/ui/editor/index.svelte:139]: @TGlide Going back to the code, I remember what I meant. If I added an early return, line 143 won't be called. 

https://github.com/TGlide/novel-svelte/blob/eae6dcdec335fc17ef86cd64b7115732022d863f/src/lib/ui/editor/index.svelte#L136-L144

Won't that be a problem, that a prop that should be called when there is a chang isn't called when `disableLocalStorage` is `true`?

Comment (TGlide) [reply src/lib/ui/editor/index.svelte:139]: True! You can move the getJson line inside the if then.

**Review by TGlide — APPROVED** [review]
thank you!

**Pull request merged**